chargeOclock/Source/Models/Client.swift

```diff
@@ -15,10 +15,15 @@ protocol ClientSettable {
 class Client {
 
 	var id: Int = 0
-	var name: String = ""
+	var isDirty: Bool = false
 
-	init(id: Int, name: String) {
-		self.id = id
+	var name: String = "" {
+		didSet {
+			isDirty = true
+		}
+	}
+
+	init(name: String) {
 		self.name = name
```

Comment:
in this case we need the id to be an optional, as we need to know if it's been set or not...
and of course, set it to a default (nil) value